### DISCUSSION TITLE
Escaping entities in inline text renderer

### DISCUSSION THREAD

**Discussion by iotch** [thephpleague/commonmark#651] [Q&A] [answered]
Hello. Is it intentional to escape everything for inline text? 

https://github.com/thephpleague/commonmark/blob/master/src/Inline/Renderer/TextRenderer.php#L36

Other inline renderers do allow HTML entities. Thanks!

**Comment by colinodell** [comment]
I'm honestly not 100% sure what the rules are for HTML entities and escaping inside of inline renderers.  But I can confirm that we're following the exact behavior of the "official" JS implementation, so what we're doing should be correct.

**Comment by colinodell** [comment, marked as answer]
I can confirm this does appear to be intentional.  `EntityParser` will match and decode any entities it finds and the `TextRenderer` will re-encode them.
